src/game/game_main.cpp

```diff
@@ -48,24 +85,65 @@ void show_int(char *buffer, void *info) {
 
 void entity_registration() {
     REGISTER_TYPE(std::vector<int>, show_buffer);
-
-    REGISTER_ENTITY(A);
-    REGISTER_ENTITY(MyEnt);
 }
 
+std::vector<Paddle *> paddles = {};
+Ball ball;
+
 void setup() {
     using namespace Input;
-    add(K(a), Name::LEFT);
-    add(K(d), Name::RIGHT);
-    add(K(w), Name::UP);
-    add(K(s), Name::DOWN);
+    add(K(a), Name::LEFT,  Player::P1);
+    add(K(d), Name::RIGHT, Player::P1);
+    add(K(w), Name::UP,    Player::P1);
+    add(K(s), Name::DOWN,  Player::P1);
+
+    add(K(j), Name::LEFT,  Player::P2);
+    add(K(l), Name::RIGHT, Player::P2);
+    add(K(i), Name::UP,    Player::P2);
+    add(K(k), Name::DOWN,  Player::P2);
+
+    Vec2 ball_points[4] = {
+        V2(-0.25, -0.25),
+        V2( 0.25, -0.25),
+        V2(-0.25,  0.25),
+        V2( 0.25,  0.25),
+    };
+    ball_shape = Physics::add_shape(LEN(ball_points), ball_points);
 
+    Vec2 paddle_points[4] = {
+        V2(-0.5, -2.5),
+        V2( 0.5, -2.5),
+        V2(-0.5,  2.5),
+        V2( 0.5,  2.5),
+    };
+    paddle_shape = Physics::add_shape(LEN(paddle_points), paddle_points);
+
+    Paddle paddle = {};
+    paddle.layer = 0;
+    paddle.position = V2(-10, 0);
+    paddle.controllable = true;
+    paddle.player = Player::P1;
+    Logic::add_entity(paddle);
+    paddles.push_back(&paddle);
+
+    Paddle paddle2 = {};
+    paddle2.layer = 0;
+    paddle2.position = V2(10, 0);
+    paddle2.controllable = true;
+    paddle2.player = Player::P2;
+    Logic::add_entity(paddle2);
+    paddles.push_back(&paddle2);
```

Comment:
Here you push the address of the paddle, but it lives on the stack so it will be invalid after this function. This is causing the crash.

Same as above.